enzyme/test/Enzyme/ReverseMode/blas/cblas_daxpy_a.ll

```diff
@@ -0,0 +1,55 @@
+;RUN: %opt < %s %loadEnzyme -enzyme -mem2reg -instsimplify -simplifycfg -S | FileCheck %s
+
+target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-apple-macosx10.17.0"
+@__const.main.m = private unnamed_addr constant [3 x double] [double 1.000000e+00, double 2.000000e+00, double 3.000000e+00], align 16
+@__const.main.d_m = private unnamed_addr constant [3 x double] [double 2.000000e+00, double 4.000000e+00, double 6.000000e+00], align 16
+@__const.main.n = private unnamed_addr constant [3 x double] [double 4.000000e+00, double 5.000000e+00, double 6.000000e+00], align 16
+@__const.main.d_n = private unnamed_addr constant [3 x double] [double 8.000000e+00, double 1.000000e+01, double 1.200000e+01], align 16
+define void @f(i32 %n, double %a, double* noalias %x, double* noalias %y) {
+entry:
+  tail call void @cblas_daxpy(i32 %n, double %a, double* %x, i32 1, double* %y, i32 1)
+  ret void
+}
+declare void @cblas_daxpy(i32, double, double*, i32, double*, i32)
+define i32 @main() {
+entry:
+  %m = alloca [3 x double], align 16
+  %d_m = alloca [3 x double], align 16
+  %n = alloca [3 x double], align 16
+  %d_n = alloca [3 x double], align 16
```

Comment:
Do you mean removing the `llvm.lifetime.start` and `llvm.lifetime.end` calls? The `.c` file only contains variable definition and a call to `__enzyme_autodiff` and these calls are generated by clang. Is there a way to strip it when generating LLVM IR?

Reply:
I actually used O2 for this, but you are absolutely right about removing the main! We don't need "real" data here and it can greatly simplify our test cases